docs/modules/pixbuf.md

```diff
@@ -0,0 +1,325 @@
+standard start and end offset Lua notation. Negative values count backwards from the end of the buffer.
+
+#### Syntax
+`buffer:shift(value [, mode[, i[, j]]])`
+
+#### Parameters
+ - `value` number of pixels by which to rotate the buffer. Positive values rotate forwards, negative values backwards.
+ - `mode` is the shift mode to use. Can be one of `pixbuf.SHIFT_LOGICAL` or `pixbuf.SHIFT_CIRCULAR`. In case of SHIFT\_LOGICAL, the freed pixels are set to 0 (off). In case of SHIFT\_CIRCULAR, the buffer is treated like a ring buffer, inserting the pixels falling out on one end again on the other end. Defaults to SHIFT\_LOGICAL.
+ - `i` is the first offset in the buffer to be affected. Negative values are permitted and count backwards from the end. Default is 1.
+ - `j` is the last offset in the buffer to be affected. Negative values are permitted and count backwards from the end. Default is -1.
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+buffer:shift(3)
+```
+
+## pixbuf.buffer:sub()
+This implements the extraction function like `string.sub`. The indexes are in leds and all the same rules apply.
+
+#### Syntax
+`buffer1:sub(i[, j])`
+
+#### Parameters
+ - `i` This is the start of the extracted data. Negative values can be used.
+ - `j` this is the end of the extracted data. Negative values can be used. The default is -1.
+
+#### Returns
+A buffer containing the extracted piece.
+
+#### Example
+```
+b = buffer:sub(1,10)
+```
+
+## pixbuf.buffer:__concat()
+This implements the `..` operator to concatenate two buffers. They must have the same number of colors per led.
+
+#### Syntax
+`buffer1 .. buffer2`
+
+#### Parameters
+ - `buffer1` this is the start of the resulting buffer
+ - `buffer2` this is the end of the resulting buffer
+
+#### Returns
+The concatenated buffer.
+
+#### Example
+```
+ws2812.write(buffer1 .. buffer2)
+```
+## pixbuf.buffer:map()
+Map a function across each pixel of one, or zip a function along two,
+pixbuf(s), storing into the buffer on which it is called.
+
+#### Syntax
+`buffer0:map(f, [buffer1], [ilo], [ihi], [buffer2, [ilo2]])`
+
+#### Parameters
+ - `f` This is the mapping function; it is applied for each pixel to each channel of `buffer1` and
+   all channels from `buffer2`, if given.  It must return a value for each channel of the output
+   buffer.
+ - `buffer1` The first source buffer.  Defaults to `buffer0`.
+ - `ilo` This is the start of the mapped range. Negative values can be used.  The default is 1.
+ - `ihi` this is the end of the mapped range. Negative values can be used. The default is -1.
+ - `buffer2` is a second buffer, for zip operations
+ - `ilo2` This is the start of the mapped range within `buffer2`.  Negative values can be used.
+   The default is 1.
+
+`buffer0` must have `ihi - ilo + 1` pixels (which is true of the defaults, when
+`buffer1` is `buffer0` and `ilo` is 1 and `ihi` is -1).
+
+#### Returns
+`buffer0`
+
+#### Example
```

Comment:
After looking at the code I understand, but from the examples and description I won't be able to use this with more than a function as param.
Especially the fact that the channels of the second buffer are also passed and can have a different number.